core/solver/gmres.cpp

```diff
@@ -341,9 +347,16 @@ void Gmres<ValueType>::apply_dense_impl(const VectorType* dense_b,
                                     span{local_num_rows * (restart_iter + 1),
                                          local_num_rows * (restart_iter + 2)},
                                     span{0, num_rhs});
-        // preconditioned_vector = get_preconditioner() * this_krylov
+        auto preconditioned_krylov = create_submatrix_helper(
```

Comment:
we could also store a std::unique_ptr outside and allocate it only if necessary?

Reply:
done